shell/plugins/dashboard-store/mutations.js

```diff
@@ -104,7 +104,9 @@ export function load(state, {
     // A specific proxy instance to used was passed in (for create -> save),
     // use it instead of making a new proxy
     entry = replaceResource(existing, data, getters);
-    addObject(cache.list, entry);
+    if (!cache.list.find((e) => e[keyField] === id)) {
```

Comment:
Can we use `cache.map`? Is there ever a case when it is in `cache.map` but not in `cache.list`? Doesn't feel like there should - so feels like we could juts check the map - this would be a lot quicker.

Checking the list feels like it would be slow for a large number of resources.

Reply:
I got very weirded out when looking at this (functionality has not change in a long while) so went down the low-bar and safest (but not performant) solution. In the near future we'll only be keeping a small amount of resources locally and it should only apply when users create new resources. I'll take another look and work out where else this (load with existing) is used and get confidence that map and list align

Reply:
I've reworked a lot of that section and added inline + review comments